application/forms/ConfigForm.php

```diff
@@ -137,8 +137,6 @@ protected function writeConfig(Config $config)
      */
     public static function transformEmptyValuesToNull(array $values)
     {
-        return array_map(function ($v) {
-            return empty($v) ? null : $v;
-        }, $values);
+        return array_map(function ($v) { return ($v === '' || (is_array($v) && empty($v))) ? null : $v; }, $values);
```

Comment:
Same as above, please check for `false` as well. And fix the codestyle, the previous style was already valid ;)